app/services/triannon/ldp_writer.rb

```diff
@@ -61,6 +62,68 @@ def self.container_exist? path
       end
     end
 
+    # Creates an empty LDP BasicContainer in LDP Storage
+    # @param [String] parent_path the path part, after the ldp base url -- in
+    #   essence, the LDP BasicContainer that will be the parent of the
+    #   to-be-created BasicContainer.
+    # @param [String] slug the value to send in Http Header 'Slug' -- this is
+    #   appended to the parent container's path to become the id of the newly
+    #   created BasicContainer
+    # @return [Boolean] true if the container was created; false otherwise
+    def self.create_basic_container parent_path, slug
+      if slug.blank?
+        puts "create_basic_container called with nil or empty slug, parent_path '#{parent_path}'"
+        return false
+      end
+      base_url = Triannon.config[:ldp]['url'].strip
+      base_url.chop! if base_url.end_with?('/')
+      slug.strip!
+      slug = slug[1..-1] if slug.start_with?('/')
+      if parent_path
+        parent_path.strip!
+        parent_path = parent_path[1..-1] if parent_path.start_with?('/')
+        parent_path.chop! if parent_path.end_with?('/')
+      end
+
+      full_path = (parent_path ? parent_path + '/' : "") + slug
+      full_url = base_url + '/' + full_path
```

Comment:
Wondering whether all the path manipulation for starting and ending '/' might be easier with something like:
```ruby
full_path = [parent_path, slug].join('/')
path_array = full_path.split('/')
full_path = path_array.join('/')
```
Or maybe the `RDF::URI.new(parent_path, slug)` will take care of everything?